src/framework/global/types/string.h

```diff
@@ -264,8 +264,11 @@ class String
     inline bool isEmpty() const { return empty(); }
     void clear();
     Char at(size_t i) const;
+    Char front() const { return at(0); }
+    Char back() const { return at(size() - 1); }
     bool contains(const Char& ch) const;
     bool contains(const String& str, CaseSensitivity cs = CaseSensitive) const;
+    bool contains(const std::wregex& re) const;
```

Comment:
Could you explain why you decided to use wregex instead of regex?
I also see that the String class used to work with std::string, now it also works with wstring. Do we want this?

Reply:
std::string - 8 bit by char
String - 16 bit by char (used std::u16string inside) 

std::regex - this is std::basic_regex<char> - 8 bit by char 

We have some symbols in the text that are 16 bits in size (various musical symbols)
When converted to std::string (UTF8), such symbols will be split into two parts, into two characters.
And accordingly, the methods `contains`, `indexOf` will not work correctly.

i.e. we need to work with UTF16, but unfortunately, ` std::basic_regex<char16_t>` is not implemented.
But `std::basic_regex<wchar_t> (std::wregex)` is implemented, where `wchar_t` is guaranteed to be greater than or equal to `char16_t`. Therefore, we can use it, forcedly, because there is no implementation with `char16_t`.